connect/eaas/core/deployment/extension.py

```diff
@@ -58,19 +58,28 @@ def deploy_extension(repo, client, log, tag=None):
     environment = extension['environments'][arguments.get('env', 'production')]
     env_api = client('devops').services[extension['id']].environments[environment['id']]
 
-    environment = stop_environment(environment, env_api, log)
-    if environment['status'] == 'running':
-        return
-
     try:
-        log(f'Creating variables for extension with package_id {arguments["package_id"]}.')
-        process_variables(arguments, env_api)
+        log(f'Processing variables for extension with package_id {arguments["package_id"]}.')
+        vars_updated = process_variables(arguments, env_api, log)
+
+        if (
+            environment.get('git', {}).get('commit') != git['commit']
+            or environment['runtime'] == 'local'
```

Comment:
In case of local you cannot stop the environment you will get an error.
In this case you should check if the status is connected and if so you have to output an error message and stop the deployment.